### PR TITLE
#2351 mandatory fields accounting bug

### PR DESCRIPTION
**Pull request by zzgvh**
# Test plan

Check that a project in the project editor can be published when all mandatory fields are filled in. Especially check the behaviour in section 8, since that's where the bug occurred. However the fix might have messed up other sections 😛 Check with the RSR validation only and both the RSR and IATI validation sets.

For everything to work the validation set for IATI also needs to be fixed: A **rsr_policymarker** validation needs to be added, set to Mandatory, and the validations for **rsr_policymarker.policy_marker** and **rsr_policymarker.significance** should also be set to Mandatory if they're not: 

screenshot_124

### CODE REVIEW

File: akvo/rsr/models/project_editor_validation.py [hunk 1]
```diff
@@ -49,14 +49,14 @@ class ProjectEditorValidation(Model):
     """
     MANDATORY_ACTION = 1
     HIDDEN_ACTION = 2
-    READ_ONLY_ACTION = 3
+    # READ_ONLY_ACTION = 3
```
Comment (punchagan) [inline akvo/rsr/models/project_editor_validation.py]: @zzgvh Is there a reason you commented this out, instead of removing the code?

**Comment by punchagan** [comment]
The discussed problem is fixed and I didn't notice any new issues. 
LGTM.

File: akvo/rsr/models/project_editor_validation.py (hunk 1 above)
Comment (zzgvh) [inline akvo/rsr/models/project_editor_validation.py]: I think we want to implement this option, so it's there as a reminder.

Comment (punchagan) [inline akvo/rsr/models/project_editor_validation.py]: I'd vote for an issue or a roadmap card with a link to this commit as a better reminder.  Sorry for the nit-pick!

File: akvo/rsr/templatetags/project_editor.py [hunk 2]
```diff
@@ -214,7 +216,7 @@ def mandatory_or_hidden(validations, field):
     """
     Retrieves the mandatory and hidden fields for project editor validations.
 
-    :returns "mandatory-1 mandatory-or-2-subtitle hidden-3"
+    :returns A string on the form mandatory-{validation ID} and/or hidden-{validation_ID}
```
Comment (punchagan) [inline akvo/rsr/templatetags/project_editor.py]: Minor typo - s/string on/string of/

File: akvo/rsr/models/project_editor_validation.py (hunk 1 above)
Comment (zzgvh) [inline akvo/rsr/models/project_editor_validation.py]: Hehe, no worries, you're right!

Comment (punchagan) [inline akvo/rsr/models/project_editor_validation.py]: Thanks!

**Comment by zzgvh** [comment]
A note for the deployment of this fix. Check that there are no validations set to Read-only. If there are any they should be removed before deploying, as they will break the admin page for project editor validation sets.

**Comment by punchagan** [comment]
#2372 has the missing migration file for this fix.

**Pull request merged**